### PR TITLE
New human corpses types

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Added new human corpses types."```

#### Purpose of change
Add more atmosphere and add possibility to tilesets authors to create tiles for different types of corpses.

#### Describe the solution
- Added 9 new corpses id with their own descriptions. They have a chance to spawn instead of a generic corpse with `A dead body.` description. The list of these corpses is hardcoded for now.
- These new corpses will spawn only as a "human" corpses, so no zombies or animals, or what else will get these corpses. Added a new `CORPSE` flag to help with this matter.
- Created a new file, `corpses.json`, that should contain all types of corpses. Moved definition of generic `corpse` and `rmi2_corpse` items there from `generic.json`.

#### Additional context
Inspired by @SomeDeadGuy's request in #33765.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -323,7 +323,29 @@ item item::make_corpse( const mtype_id &mt, time_point turn, const std::string &
         debugmsg( "tried to make a corpse with an invalid mtype id" );
     }
 
-    item result( "corpse", turn );
+    std::string corpse_type;
+
+    if( mt == mtype_id::NULL_ID() ) {
+        //TODO: Jsonify this set.
+        static const std::set<std::string> corpse_types = {
```
Comment (BevapDin) [inline src/item.cpp]: Suggestion: make this an item group.

**Comment by SourSlime** [comment]
Are the corpse descriptions meant to have gender identifiers like they currently do?

**Comment by Night-Pryanik** [comment]
> Are the corpse descriptions meant to have gender identifiers like they currently do?

Yes, that was the original plan.

**Comment by Night-Pryanik** [comment]
Well, I managed to use item group to determine what corpse type should spawn. This solves two problems - jsonified list instead of a hardcoded one, and weighted list instead of equal chances for every type of corpse.
There is one issue though - for some reason game doesn't use new values from json for material, weight and volume, and all corpses spawn with equal parameters. I'd appreciate some help with this matter.

File: data/json/items/corpses.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    "description": "It's hard to tell how exactly did this person die, but judging from his twisted body and frightful grimace, it was a really painful death.",
+    "looks_like": "corpse",
+    "material": "hflesh"
+  },
+  {
+    "type": "GENERIC",
+    "id": "corpse_scorched",
+    "copy-from": "corpse",
+    "name": "corpse",
+    "description": "An awful mass of charred flesh.  You're not even sure if it's a man or a woman.  Judging by the stain of vaporized blood next to the body, he probably was shot prior to the burning.",
```
Comment (FuelType-Memes) [inline data/json/items/corpses.json]: "Not sure if it's a man or a woman" maybe change "he probably" to "they probably"?

File: data/json/items/corpses.json [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    "description": "A dead body with a big hole in the back of its neck.  Looks like someone backstabbed this fellow.  You can only guess whether he deserved it.",
+    "looks_like": "corpse",
+    "material": "hflesh"
+  },
+  {
+    "type": "GENERIC",
+    "id": "corpse_gunned",
+    "copy-from": "corpse",
+    "name": "corpse",
+    "description": "One might say that face of this woman is devoid of flaws. That could be true if there wasn't this big ugly hole right in the center of her forehead, and even bigger hole in her hindhead.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/corpses.json]: This 'her hindhead' is a bit of an oddity. Perhaps 'the back of her head' would sound better?

**Comment by kilozombie** [comment]
Copy of my question that I put on the previous corpse-related PR, just a question of why human corpses should be much more prevalent than at present.

>Could I ask how this fits into the lore?

>It's been a while, and I heard there's been a lore rewrite, but my knowledge of Cataclysm is that a vast majority of the human population became Z because the blob was in the water grid long before Hour One. The only reason that scientist, military, drug dealer corpse map extras existed was because they were carrying bottled water exempt from the blob that most other people ingested.

>That's why there are hundreds of Zs in town-- it didn't matter how you died, you were already fated to become one.

**Comment by Night-Pryanik** [comment]
I answered your question in that PR, no need to duplicate it here.

**Pull request merged**